### PR TITLE
[TypeReconstruction] Apply substistution when reconstructing typealia…

### PR DESCRIPTION
**Pull request by dcci**
…ses.

Thanks to Slava for pointing this fix out.

### CODE REVIEW

**Comment by dcci** [comment]
Paired with a test in lldb: https://github.com/apple/swift-lldb/pull/662

**Comment by dcci** [comment]
@swift-ci please smoke test and merge

File: lib/IDE/TypeReconstruction.cpp [hunk 1]
```diff
@@ -792,11 +792,11 @@ static void VisitNodeGenericTypealias(ASTContext *ast,
...
   if (auto nominal = genericTypeAlias->getDeclContext()
                          ->getAsNominalTypeOrNominalTypeExtensionContext()) {
-    parentType = nominal->getDeclaredInterfaceType();
+    parentType = nominal->getDeclaredInterfaceType().subst(subMap);
   }
-  NameAliasType *NAT =
-      NameAliasType::get(genericTypeAlias, parentType, subMap,
-                         genericTypeAlias->getUnderlyingTypeLoc().getType());
+  NameAliasType *NAT = NameAliasType::get(
+      genericTypeAlias, parentType, subMap,
```
Comment (slavapestov) [inline lib/IDE/TypeReconstruction.cpp]: The parent type is still wrong, it should have the same substitution map applied

**Comment by slavapestov** [comment]
Thanks!

**Pull request merged**